### PR TITLE
Add RFC 6901 support.

### LINKED ISSUE #249: Question/RFE: support for RFC 6901: JSON Pointer
Hey Jacob,

I came across [RFC 6901: JSON pointer](https://tools.ietf.org/html/rfc6901) and was wondering if JSON-Fortran has support for this or would be interested in supporting this. i.e. I'm wondering if the current syntax for addressing elements follows this RFC 6901.... or what would need to happen to implement it. (See the example section)

### PR DESCRIPTION
**Pull request by jacobwilliams**
Fixes #249

Also some minor changes: `initialize` arguments now in an include file.
Also fixed some documentation link issues.

### CODE REVIEW

**Comment by jacobwilliams** [comment]
Something weird happening on Travis. Seems to be crashing during the coverage generation?

**Comment by zbeekman** [comment]
I'm guessing that it's either https://github.com/jacobwilliams/json-fortran/blob/d3f70c0df2cfa3ea49027f069bb76b76ca885dec/build.sh#L348 or https://github.com/jacobwilliams/json-fortran/blob/d3f70c0df2cfa3ea49027f069bb76b76ca885dec/build.sh#L353 where the error is occurring... I can't remember how we set up gcov, but with other projects I've had to ensure that gcov matches the gfortran/gcc version so if I compile with `gfortran-6` I invoke `gcov-6`

**Comment by jacobwilliams** [comment]
I think that's taken care of here: https://github.com/jacobwilliams/json-fortran/blob/master/.travis.yml#L59
It always worked before, not sure what's changed.

**Comment by jacobwilliams** [comment]
The error is here:
```
+./pages/development-resources/gccr.pl -n -c json_value_module.F90-no-unicode.gcov no-unicode json_value_module.F90-unicode.gcov unicode
ERROR: cannot parse line 7241 in file json_value_module.F90-no-unicode.gcov
 Is this a valid gcov file?
```
I don't see anything unusual on this line though. See [raw Travis log](https://api.travis-ci.org/jobs/198723183/log.txt?deansi=true).

**Comment by zbeekman** [comment]
I guess line 7241 of the gcov file corresponds to line 7236 of son_value_module.F90 (no unicode, so perhaps haver preprocessing) since there are 5 lines at the beginning of the gcov file before the hit count of the source code begins. I'm wondering if gccr.pl hasn't allocated a big enough string to read the git count? Or a big enough integer? Just guesses, since 7236 looks completely kosher to me.

**Comment by zbeekman** [comment]
One possible solution, is to ditch gccr and and split the build matrix to have a separate case for unicode and non-unicode.... then codecov will merge the coverage info. I'm wondering if GCCR is a remnant from the coverall days, or why we didn't take this approach before... it has the potential to simplify things.

**Comment by zbeekman** [comment]
> One possible solution, is to ditch gccr and and split the build matrix to have a separate case for unicode and non-unicode.... then codecov will merge the coverage info. I'm wondering if GCCR is a remnant from the coverall days, or why we didn't take this approach before... it has the potential to simplify things.

Although I don't remember the intricacies of the FORD documentation and whether or not we need things from both builds... but I'm guessing we don't.

**Comment by jacobwilliams** [comment]
I'm OK with whatever you want to do. :)

I submitted a ticket for gccr. Maybe there is a quick fix (I know nothing about perl...)

**Comment by zbeekman** [comment]
> I submitted a ticket for gccr. Maybe there is a quick fix (I know nothing about perl...)

The only thing I know about perl is that I hate it.... maybe I can look through the source and post a question on stack overflow or something like that.... Another dirty trick might be to just truncate large hit counts since we don't care about how times a line was hit, beyond a point. You could just send the output through `sed` on travis.yml when there are more than N integers you truncate to M. Then we could see if that is the source of the issue.

**Comment by jacobwilliams** [comment]
I guess I'm a perl wizard and never knew it!

**Comment by jacobwilliams** [comment]
I think this feature is finished. To use it, you have to set `use_rfc6901_paths` to `.true.` in the appropriate `initialize` routine. Then any path is interpreted as an RFC 6901 "JSON pointer". By default, the old way is used to preserve backward compatibility.

File: pages/development-resources/gccr.pl [hunk 1]
```diff
@@ -248,7 +248,7 @@ ($$)
 			if($save_raw) {
 				$data{'line'}[$file_line_num]{'raw'} = $raw;
 			}
-		}elsif($line =~ /^\s+(\d+):\s*(\d+):(.*)/) {
+		}elsif($line =~ /^\s*(\d+):\s*(\d+):(.*)/) {
```
Comment (zbeekman) [inline pages/development-resources/gccr.pl]: yep, that's definitely the culprit! gcov is producing a line with no leading space. They should patch this upstream.... if there is an upstream

**Pull request merged**